src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -147,6 +155,46 @@ public void testRunRequestAsync() throws Exception {
     }
 
     @Test
+    public void testRunRequestWithSocketTimeout() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri);
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
+
+        stubFor(post(urlEqualTo("/login"))
+                .withHeader("Accept", equalTo("application/json"))
+                .withHeader("Content-Type", equalTo("application/json"))
+                .willReturn(aResponse()
+                .withFixedDelay(2000)));
+
+        clientWithFastTimeout.login("user", "pass");
+    }
+
+    @Test
+    public void testRunRequestWithSocketTimeoutThroughJDKConnection() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri, new JDKConnectionFactory());
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
```

Comment:
This two tests check that the "Read timed out" is well implemented, that's why I force a short timeout of 1s, and delay the server response of 2s, to catch and valid the "Read timed out" error. This is just an hack to quickly pass tests. If we want to test with the default config, we have to delay the response more than 10000ms, twice... 

Is it OK ?